### PR TITLE
base-files: reduce IPv6 ULA prefix generation to a single call

### PR DESCRIPTION
**Pull request by systemcrash**
Tested on: 23.05.3

### CODE REVIEW

File: package/base-files/files/etc/uci-defaults/12_network-generate-ula
Comment (rany2) [inline package/base-files/files/etc/uci-defaults/12_network-generate-ula:3]: This could be improved further, please do: `hexdump -vn 5 -e '5/1 "%02x"' /dev/urandom`.

But overall, I agree with the spirit of this change!

**Review by rany2 — CHANGES_REQUESTED** [review]

File: package/base-files/files/etc/uci-defaults/12_network-generate-ula
Comment (champtar) [reply package/base-files/files/etc/uci-defaults/12_network-generate-ula:3]: Aren't we super early in the boot and /dev/urandom is likely not super random ?

Comment (rany2) [reply package/base-files/files/etc/uci-defaults/12_network-generate-ula:3]: I'm not sure how important that is seeing this for IPv6 ULA, it's probably random enough. If it ever became a problem maybe something like `dd if=/dev/urandom bs=1000 count=1 2>/dev/null | sha256sum` would work fine.

Comment (systemcrash) [reply package/base-files/files/etc/uci-defaults/12_network-generate-ula:3]: A partial consideration for this change was that entropy might be starved at boot on some platforms, but I figured running this script means we're already past boot.

Comment (systemcrash) [reply package/base-files/files/etc/uci-defaults/12_network-generate-ula:3]: > This could be improved further, please do: `hexdump -vn 5 -e '5/1 "%02x"' /dev/urandom`.
> 

That's quite good - though not everyone is familiar with hexdump, whereas dd is more ubiquitous, and easy to understand what's being piped to hexdump, at least.

Comment (systemcrash) [reply package/base-files/files/etc/uci-defaults/12_network-generate-ula]: Amended.

Comment (systemcrash) [reply package/base-files/files/etc/uci-defaults/12_network-generate-ula]: Welp, if you're not merging this, you can at least approve it. 🙏

**Review by rany2 — APPROVED** [review]

**Review by ehem — APPROVED** [review]
Seems ridiculous to need an entire pull request for a single ridiculously simple bit of cleanup.  There is always so much one would tend to want to batch cleanup together.

Anyway, cleanup is always a Good Thing.

**Comment by systemcrash** [comment]
Yeah, thanks @ehem where does one draw the line size wise? This repo needs like 10x more people with commit access to manage the volume, it seems. So one would think that two lines of change would be trivial. 🤷

**Comment by Shine-** [comment]
Elegant solution.
I'm using this one-liner in my scripts:
```
uci set network.globals.ula_prefix="$(head -c 5 /dev/urandom | hexdump -e '"fd" 5/1 "%02x"' | sed "s/..../&:/g")
:/48"
```
But yours even saves my additional `busybox` call caused by using `sed`. I generally avoid messing with IFS, though, out of personal preference :-)

**Comment by Shine-** [comment]
No credit needed or wanted. You can treat all my contributions as Public Domain.

File: package/base-files/files/etc/uci-defaults/12_network-generate-ula
Comment (Shine-) [inline package/base-files/files/etc/uci-defaults/12_network-generate-ula:4]: The `2/2 "%02x:"` part should be either `2/2 "%x:"` instead, for compressed IPv6 notation (preferred, e.g. `fd01:2:345::/48`) or `2/2 "%04x:"` for uncompressed notation (`fd01:0002:0345::/48`).

**Review by ehem — APPROVED** [review]
Starting to get a bit funky with `hexdump` here.  I though still think this is a _distinct_ improvement to the script.

**Comment by ehem** [comment]
Usually one starts wrapping lines near the 80 column mark.  Mainly the traditional size of a dumb terminal screen.

The issue with reviewers is their motivation and this usually boils down to money in some form.  I've see projects which handle larger volumes of commits with fewer reviewers than OpenWRT has.  Those reviewers had employers with financial interest in those projects being high quality.

Here I'm guessing the incentives for people with commit access aren't the greatest.  I don't know what the issues are, nor do I have any idea of what to correct.

**Comment by systemcrash** [comment]
> Starting to get a bit funky with `hexdump` here. I though still think this is a _distinct_ improvement to the script.

haha. Give it like another 3 moths and maybe we can get it down to 7 bytes or something. :D

**Comment by systemcrash** [comment]
> Usually one starts wrapping lines near the 80 column mark. Mainly the traditional size of a dumb terminal screen.

Yeah, how aggressive should we be here. There's always
```
uci set blah=\
"next line"
```

**Comment by ehem** [comment]
Always varies depending upon reviewer and author.  Sometimes there isn't any good spot to break the line and simply continuing is best.  Most of the rules are "rules of thumb".

**Review by rany2 — APPROVED** [review]
Pretty impressive how this PR just keeps getting more consise, but surely this is its final form :D

**Comment by Ansuel** [comment]
dbz-goku

Very tempted to not merge this and see if it can be improved even more.

And imagine me coming and NACK this as it's not readable :D (just joking)

**Pull request merged**
